tests/Repository/DefaultUnleashRepositoryTest.php

```diff
@@ -168,6 +168,33 @@ public function testCacheWithNoFeatures()
         self::assertNull($repository->findFeature('test'));
     }
 
+    public function testRefreshCache()
+    {
+        $cache = $this->getRealCache();
+        $repository = new DefaultUnleashRepository(
+            new Client([
+                'handler' => HandlerStack::create($this->mockHandler),
+            ]),
+            new HttpFactory(),
+            (new UnleashConfiguration('', '', ''))
+                ->setCache($cache)
+                ->setTtl(5)
+        );
+
+        $this->pushResponse($this->response, 3);
+        $repository->getFeatures();
+
+        sleep(3);
+        $repository->refreshCache();
+        sleep(3);
+
+        self::assertEquals(1, $this->mockHandler->count());
+
+        //Cache entry should have been invalidated by now (3+3 > TTL) if we didn't refresh it
+        $repository->getFeatures();
+        self::assertEquals(1, $this->mockHandler->count());
```

Comment:
Is the count here not cumulative? As in, weren't two http requests made during this test?

Reply:
I also was a bit puzzled, but the `count()` method is returning the remaining items in the queue. You could argue to start with 2 response in the queue and check for 0 here...

```
    /**
     * Returns the number of remaining items in the queue.
     */
    public function count(): int
    {
        return \count($this->queue);
    }
```

Reply:
That would be better IMO, this got me all confused. You can also use `assertCount($expectedAmount,$this->requestHistory)`.

Reply:
Changed